src/Interpreters/ServerAsynchronousMetrics.cpp

```diff
@@ -106,6 +108,26 @@ void ServerAsynchronousMetrics::updateImpl(TimePoint update_time, TimePoint curr
             "Total number of entries in the userspace page cache." };
     }
 
+#if defined(OS_LINUX) || defined(OS_FREEBSD)
+    {
+        MemoryStatisticsOS memory_stat;
+        MemoryStatisticsOS::Data memory_data = memory_stat.get();
```

Comment:
This is already done in `AsynchronousMetrics.cpp` to assign `MemoryResident` metric. Can't the new metric be added there, right next to `MemoryResident`, instead of querying the OS twice?

Reply:
Can just omit that stat if `context->getPageCache()` is null?

Reply:
I was living in pre- #65910 era, where we have separate `Context` for standalone build.

Also we can use value installed by the `AsynchronousMetrics` - `new_values["MemoryResident"]` here

But I'm with any option